kernel/kernel-api/src/main/java/io/delta/kernel/engine/CommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.engine;
+
+import java.io.IOException;
+import java.util.Map;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.coordinatedcommits.*;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractMetadata;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractProtocol;
+import io.delta.kernel.utils.CloseableIterator;
+
+/**
```

Comment:
general comment to all APIs in this interface: what exception is thrown if the table is not managed by a coordinated commit?

Reply:
I think the commit coordinator assumes that the logPath passed as an argument represents a table managed by the commit coordinator. For example, the [InMemoryCommitCoordinator](https://github.com/delta-io/delta/blob/fb4d88d1e4a762ffb81adb10f7801784ec24bf97/spark/src/main/scala/org/apache/spark/sql/delta/coordinatedcommits/InMemoryCommitCoordinator.scala#L125) in Delta Spark doesn't throw a specific exception for this case (will just throw a NullPointerException). So the user should ensure that they pass a correct logPath.
@dhruvarya-db Do I understand correctly?

Reply:
I don't think that there has been any standardization around that. @prakharjain09 any idea?

Reply:
No standardization for now.